pupil-spa/src/app/services/check-recovery/check-recovery.service.spec.ts

```diff
@@ -0,0 +1,37 @@
+import { TestBed } from '@angular/core/testing';
+
+import { StorageService } from '../storage/storage.service';
+import { CheckRecoveryService } from './check-recovery.service';
+import * as auditFailedResponseMock from '../check-recovery/audit.failed.response.mock.json';
+import * as auditSuccessResponseMock from '../check-recovery/audit.success.response.mock.json';
+let storageService: StorageService;
+let checkRecoveryService: CheckRecoveryService;
+
+describe('CheckRecoveryService', () => {
+  beforeEach(() => {
+    const inject = TestBed.configureTestingModule({
+        providers: [
+          StorageService,
+          CheckRecoveryService
+        ]
+      }
+    );
+    storageService = inject.get(StorageService);
+    checkRecoveryService = inject.get(CheckRecoveryService);
+  });
+  it('hasUnfinishedCheck should return true if there is no CheckComplete entry in the audit', () => {
```

Comment:
would it not be cleaner to determine if the check has been submitted simply by finding an entry in localStorage, or do you feel this would be more deterministic?

Reply:
@GuyHarwood I was discussing this idea of having an extra (boolean) property in the local storage to verify whether a check is submitted or not earlier today with @jon-shipley
I have the impression that this property will still rely on the audit log eventually so possibly wouldn't provide some added value apart from being a shortcut.

Reply:
sounds like a duplicate way of determining state, which just makes the system less robust.

if the check has been submitted, surely the local storage payload will be gone?

Reply:
Currently the local storage is being wiped out upon successful new pupil login.
Meaning that the redirect will only happen if the audit array (which still is present at this stage) is populated and the type 'CheckComplete' is missing.
Aside from relying in the audit section of the local storage would there be a better approach that comes in mind?

Reply:
i would have thought that this made the most sense.....
### Happy Path
- User completes check
- Check is submitted successfully
- Check data is wiped from local storage

### Unhappy Path
- User completes check
- Check is not submitted successfully 
- Check remains in memory
- Next user attempts to logon
- Previous check is discovered
- Previous check is uploaded

Reply:
@GuyHarwood updated as per our discussion